File: docs/src/testers/functions/invocations/disable_raise.ts
// @sniptest filename=disable_raise.ts
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();
const fn = client.NotteFunction({
  function_id: process.env.NOTTE_FUNCTION_ID!,
});

// Inspect a failed run without throwing an exception.
const result = await fn.run(
  { url: 'https://example.com', fail: true },
  { raiseOnFailure: false },
);
console.log(JSON.stringify({ status: result.status }));

File: docs/src/testers/functions/invocations/disable_raise.py
# @sniptest filename=disable_raise.py
import json
import os

from notte_sdk import NotteClient

client = NotteClient()
function = client.Function(os.environ["NOTTE_FUNCTION_ID"])

# Inspect a failed run without raising an exception.
result = function.run(url="https://example.com", fail=True, raise_on_failure=False)
print(json.dumps({"status": result.status}))
